buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -145,13 +148,15 @@ enum MagazineCaching {
                 @Override
                 protected Object initialValue() {
                     if (cachedMagazinesNonEventLoopThreads || ThreadExecutorMap.currentExecutor() != null) {
-                        Magazine mag = new Magazine(AdaptivePoolingAllocator.this, false);
-
-                        if (FastThreadLocalThread.willCleanupFastThreadLocals(Thread.currentThread())) {
-                            // Only add it to the liveMagazines if we can guarantee that onRemoval(...) is called,
-                            // as otherwise we might end up holding the reference forever.
-                            liveMagazines.add(mag);
+                        if (!FastThreadLocalThread.willCleanupFastThreadLocals(Thread.currentThread())) {
+                            // To prevent potential leak, we will not use thread-local magazine.
+                            logger.warn("Thread-local magazine will NOT be used, since the current thread:{} " +
```

Comment:
is warn really the right level here ?

Reply:
Honestly, I don't think we need any logging.